File: src/designer/designer_packages/elsa-workflows-designer/src/modules/switch/flow/flow-switch-port-provider.ts
import 'reflect-metadata';
import {Service} from "typedi";
import {Activity, Port, PortType} from "../../../models";
import {FlowSwitchActivity} from "./models";
import {PortProvider, PortProviderContext} from "../../../services";

@Service()
export class FlowSwitchPortProvider implements PortProvider {

  getOutboundPorts(context: PortProviderContext): Array<Port> {
    const activity = context.activity as FlowSwitchActivity;

    if (activity == null)
      return [];

    const cases = activity.cases ?? [];
    const ports = cases.map(x => ({name: x.label, displayName: x.label, type: PortType.Flow}));
    const defaultPort: Port = {name: 'Default', displayName: 'Default', type: PortType.Flow};
    return [...ports, defaultPort];
  }

  resolvePort(portName: string, context: PortProviderContext): Activity | Array<Activity> {
    return null;
  }

  assignPort(portName: string, activity: Activity, context: PortProviderContext) {
    return null;
  }
}

File: src/studio/modules/Elsa.Studio.ActivityPortProviders/Providers/FlowSwitchPortProvider.cs
using System.Text.Json;
using Elsa.Api.Client.Extensions;
using Elsa.Api.Client.Resources.ActivityDescriptors.Enums;
using Elsa.Api.Client.Resources.ActivityDescriptors.Models;
using Elsa.Api.Client.Shared.Models;
using Elsa.Studio.Workflows.Domain.Contexts;
using Elsa.Studio.Workflows.Domain.Providers;
using JetBrains.Annotations;

namespace Elsa.Studio.ActivityPortProviders.Providers;

/// <summary>
/// Provides ports for the FlowSwitch activity based on its cases.
/// </summary>
[UsedImplicitly]
/// <summary>
/// Provides flow switch port services.
/// </summary>
public class FlowSwitchPortProvider : ActivityPortProviderBase
{
    /// <inheritdoc />
    public override bool GetSupportsActivityType(PortProviderContext context) => context.ActivityDescriptor.TypeName is "Elsa.FlowSwitch";

    /// <inheritdoc />
    public override IEnumerable<Port> GetPorts(PortProviderContext context)
    {
        var options = new JsonSerializerOptions
        {
            PropertyNamingPolicy = JsonNamingPolicy.CamelCase
        };

        var cases = context.Activity.GetProperty<List<SwitchCase>>(options, "cases") ?? new List<SwitchCase>();

        foreach (var @case in cases)
        {
            yield return new Port
            {
                Name = @case.Label,
                DisplayName = @case.Label,
                Type = PortType.Flow,
            };
        }

        if (cases.Any(x => x.Label == "Default"))
            yield break;

        yield return new Port
        {
            Name = "Default",
            DisplayName = "Default",
            Type = PortType.Flow
        };
    }
}